Working towards a much less brittle `test.py`

h4110w33n opened the pull request:
There are a lot of things being checked, but most are either data type or just size >0. This way we don't run into issues with getting hyper-specific.

h4110w33n committed:
Working towards a much less brittle `test.py`

There are a lot of things being checked, but most are either data type or just size >0. This way we don't run into issues with getting hyper-specific

h4110w33n commented:
#9

h4110w33n committed:
The bulk of the work

* Just about everything is now testing in a sane fashion.
* Linting of the `test.py` is now a thing.
* Docs have been updated to indicate that the aliases aren't explicitly tested since they are aliases.
* `requirements.txt` have been updated to include Pandas.

h4110w33n committed:
As of the v2 release for the community, legacy v1 api is deprecated.

https://coinmetrics.atlassian.net/servicedesk/customer/portal/2/article/339968003

	Note that Community API is not available for API version 1: please use version 2 or higher.

h4110w33n committed:
Pulling the proAPI components that negativly impact coverage

h4110w33n committed:
fixing requirements

h4110w33n committed:
pluralization will be the death of me

h4110w33n committed:
adding the pep8 speaks checker

pep8speaks commented:
Hello @h4110w33n! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:



* In the file [`coinmetrics/community.py`](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/coinmetrics/community.py):

> [Line 9:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/coinmetrics/community.py#L9): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 177:9](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/coinmetrics/community.py#L177): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '

* In the file [`docs/conf.py`](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/docs/conf.py):

> [Line 17:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/docs/conf.py#L17): [E402](https://duckduckgo.com/?q=pep8%20E402) module level import not at top of file
> [Line 55:34](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/docs/conf.py#L55): [W292](https://duckduckgo.com/?q=pep8%20W292) no newline at end of file

* In the file [`setup.py`](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py):

> [Line 13:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L13): [E101](https://duckduckgo.com/?q=pep8%20E101) indentation contains mixed spaces and tabs
> [Line 13:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L13): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 13:2](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L13): [E117](https://duckduckgo.com/?q=pep8%20E117) over-indented
> [Line 16:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L16): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 17:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L17): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 18:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L18): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 19:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L19): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 20:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L20): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 21:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L21): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 21:101](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L21): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (102 > 100 characters)
> [Line 22:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L22): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 23:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L23): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 24:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L24): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 25:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L25): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 26:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L26): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 27:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L27): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 28:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L28): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 29:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L29): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 30:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L30): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 31:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L31): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 32:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L32): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 33:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L33): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 34:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L34): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 35:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L35): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 36:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L36): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 37:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L37): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 38:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L38): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 39:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L39): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 40:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L40): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 41:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L41): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 42:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L42): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 43:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L43): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 44:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L44): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 45:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L45): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 46:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L46): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs
> [Line 47:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/setup.py#L47): [W191](https://duckduckgo.com/?q=pep8%20W191) indentation contains tabs

* In the file [`test.py`](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/test.py):

> [Line 44:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/test.py#L44): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 151:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/test.py#L151): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 227:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/test.py#L227): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 492:9](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/test.py#L492): [E265](https://duckduckgo.com/?q=pep8%20E265) block comment should start with '# '
> [Line 561:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/test.py#L561): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 660:1](https://github.com/h4110w33n/coinmetrics/blob/f82f07026f8e942d70e6c8df48c7034cb24b1afd/test.py#L660): [E305](https://duckduckgo.com/?q=pep8%20E305) expected 2 blank lines after class or function definition, found 1



##### Comment last updated at 2019-11-13 02:43:09 UTC

h4110w33n committed:
adjustements for pep8speaks

h4110w33n committed:
tweaks for codecov

h4110w33n committed:
adding codecov badge to readme

h4110w33n committed:
adding correct spacting

h4110w33n committed:
adjusting codecov to get everything, not just diff for now

h4110w33n commented:
There is still work to do to consolidate the 'version' that is used by sphinx and twine so we don't need to update it in two places.

h4110w33n committed:
adding travis ci refinements on windows

h4110w33n committed:
more windows refinements

h4110w33n committed:
consolidated version data

Moving the 'source of truth' for the package version to the
module itself. We should now be compliant with PEP396, and
the features it facilitates

h4110w33n committed:
that empty new line

h4110w33n committed:
Merge branch 'master' into v2-better-testing

h4110w33n committed:
removing trailing whitespace

h4110w33n committed:
Merge branch 'v2-better-testing' of github.com:h4110w33n/coinmetrics into v2-better-testing

h4110w33n committed:
old version got pulled back in by accidant.

h4110w33n committed:
build fixes are in place. we should be gtg

h4110w33n commented:
> There is still work to do to consolidate the 'version' that is used by sphinx and twine so we don't need to update it in two places.

The version is now located in the `__init__.py` file under `__version__` and should now be compatible with PEP396. 

Aiming to give this a once-over tomorrow and merge.
